qiskit/providers/ibmq/api_v2/session.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Session customized for IBM Q access."""
+
+from requests import Session, RequestException
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from .exceptions import ApiError
+
+
+STATUS_FORCELIST = (
+    500,  # Internal Server Error
+    502,  # Bad Gateway
+    503,  # Service Unavailable
+    504,  # Gateway Timeout
+)
+CLIENT_APPLICATION = 'qiskit-api-py'
+
+
+class RetrySession(Session):
+    """Session with retry and handling of IBM Q parameters."""
```

Comment:
It was a little bit difficult to know what was going on here but I found this article "[Best practices with retries and requests](https://www.peterbe.com/plog/best-practice-with-retries-with-requests)" which sheds some light. Leaving it here for the record.

Could you explain in some more detail how do `Session`, `HTTPAdapter` and `Retry` play together?

Reply:
I added a generic note in #82 - in the same spirit of the other comment, I think we can assume that contributors interested in modifying the class will refer to the `requests` documentation directly after seeing the class itself. If not, we can review (actually might be a good idea to have another pair of eyes documenting it, in case is not clear enough)!